e2e/README.md

```diff
@@ -0,0 +1,315 @@
+
+
+### How tests are run
+
+The tests use the `matrix` feature of Github Actions. The matrix is
+dynamically generated using [this script](https://github.com/cosmos/ibc-go/blob/main/.github/scripts/build_test_matrix.go).
+
+> Note: there is currently a limitation that all tests belonging to a test suite must be in the same file.
+In order to support test functions spread in different files, we would either need to manually maintain a matrix
+or update the script to account for this. The script assumes there is a single test suite per test file to avoid an overly complex
+generation process.
+
+Which looks under the `e2e` directory, and creates a task for each test suite function.
+
+#### Example
+
+```go
+// e2e/file_one_test.go
+package e2e
+
+func TestFeeMiddlewareTestSuite(t *testing.T) {
+	suite.Run(t, new(FeeMiddlewareTestSuite))
+}
+
+type FeeMiddlewareTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+func (s *FeeMiddlewareTestSuite) TestA() {}
+func (s *FeeMiddlewareTestSuite) TestB() {}
+func (s *FeeMiddlewareTestSuite) TestC() {}
+
+```
+
+```go
+// e2e/file_two_test.go
+package e2e
+
+func TestTransferTestSuite(t *testing.T) {
+	suite.Run(t, new(TransferTestSuite))
+}
+
+type TransferTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+func (s *TransferTestSuite) TestD() {}
+func (s *TransferTestSuite) TestE() {}
+func (s *TransferTestSuite) TestF() {}
+
+```
+
+In the above example, the following would be generated.
+
+```json
+{
+   "include": [
+      {
+         "suite": "FeeMiddlewareTestSuite",
+         "test": "TestA"
+      },
+      {
+         "suite": "FeeMiddlewareTestSuite",
+         "test": "TestB"
+      },
+      {
+         "suite": "FeeMiddlewareTestSuite",
+         "test": "TestC"
+      },
+      {
+         "suite": "TransferTestSuite",
+         "test": "TestD"
+      },
+      {
+         "suite": "TransferTestSuite",
+         "test": "TestE"
+      },
+      {
+         "suite": "TransferTestSuite",
+         "test": "TestF"
+      }
+   ]
+}
+```
+
+This string is used to generate a test matrix in the Github Action that runs the E2E tests.
+
+All tests will be run on different hosts.
+
+
+#### Misceleneous:
+
+* Gas fees are set to zero to simply calcuations when asserting account balances.
+* When upgrading from e.g. v4 -> v5, in ibc-go, we cannot upgrade the go.mod under `e2e` since v5 will not yet exist. We need to upgrade it in a follow up PR.
+
+### Troubleshooting
+
+* On Mac, after running a lot of tests, it can happen that containers start failing. To fix this, you can try clearing existing containers and restarting the docker daemon.
+
+  This generally manifests itself as relayer or simd containers timing out during setup stages of the test. This doesn't happen in CI.
+  ```bash
+  # delete all images
+  docker system prune -af
+  ```
+  This issue doesn't seem to occur on other operating systems.
+
```

Comment:
Should these be indented?

Reply:
This was intentional as all this text is relevant to the first (and only) bullet point